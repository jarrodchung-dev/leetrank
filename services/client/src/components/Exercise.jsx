import React from "react";
import AceEditor from "react-ace";
import "ace-builds/src-noconflict/mode-python";
import "ace-builds/src-noconflict/mode-javascript";
import "ace-builds/src-noconflict/theme-github";

const Exercise = (props) => {
  return (
    <>
      <span style={{ fontSize: "95%" }}>{props.exercise.body}</span>
      <AceEditor
        mode="python"
        theme="github"
        name={props.exercise.id.toString()}
        fontSize={12}
        height={"220px"}
        showPrintMargin={true}
        showGutter={true}
        highlightActiveLine={true}
        value={props.editor.value}
        style={{ marginBottom: "10px", marginTop: "10px" }}
        editorProps={{ $blockScrolling: Infinity }}
        onChange={props.handleChange}
      />
      {props.isAuthenticated && (
        <div>
          <button
            className="button"
            onClick={(event) => props.submitExercise(event, props.exercise.id)}
            disabled={props.editor.button.isDisabled}
          >
            Run Code
          </button>
          <br />
          <br />
          <div className="level">
            {props.editor.showGrading && (
              <h5 className="has-text-left">
                <span className="grade-text">Grading...</span>
              </h5>
            )}
            {props.editor.showCorrect && (
              <h5 className="has-text-left">
                <span className="grade-text">Correct!</span>
              </h5>
            )}
            {props.editor.showIncorrect && (
              <h5 className="has-text-left">
                <span className="grade-text">Incorrect!</span>
              </h5>
            )}
          </div>
        </div>
      )}

      <br />
    </>
  );
};

export default Exercise;
