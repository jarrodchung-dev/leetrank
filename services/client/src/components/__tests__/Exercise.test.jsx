import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import AceEditor from "react-ace";
import Exercise from "../Exercise.jsx";

jest.mock("react-ace");

beforeEach(() => {
  console.error = jest.fn();
  console.error.mockClear();
});

describe("Exercise", () => {
  const exercise = {
    id: 0,
    body:
      "Define a function called sum that takes two arguments as arguments and returns their sum"
  };
  const editor = {
    value: "",
    button: { isDisabled: false },
    showGrading: false,
    showCorrect: false,
    showIncorrect: false
  };

  it("renders properly", () => {
    const wrapper = shallow(
      <Exercise exercise={exercise} editor={editor} isAuthenticated={false} />
    );
    const span = wrapper.find("span");
    expect(span.length).toBe(1);
    expect(span.text()).toBe(exercise.body);
  });

  it("renders a snapshot properly when a user is authenticated", () => {
    let isAuthenticated = true;
    const tree = renderer
      .create(<Exercise exercise={exercise} editor={editor} isAuthenticated={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a snapshot properly when not authenticated", () => {
    const tree = renderer
      .create(<Exercise exercise={exercise} editor={editor} isAuthenticated={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
