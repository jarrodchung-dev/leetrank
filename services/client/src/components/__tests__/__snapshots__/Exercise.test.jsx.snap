// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise renders a snapshot properly when a user is authenticated 1`] = `
Array [
  <span
    style={
      Object {
        "fontSize": "95%",
      }
    }
  >
    Define a function called sum that takes two arguments as arguments and returns their sum
  </span>,
  <div>
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
    >
      Run Code
    </button>
    <br />
    <br />
    <div
      className="level"
    />
  </div>,
  <br />,
]
`;

exports[`Exercise renders a snapshot properly when not authenticated 1`] = `
Array [
  <span
    style={
      Object {
        "fontSize": "95%",
      }
    }
  >
    Define a function called sum that takes two arguments as arguments and returns their sum
  </span>,
  <br />,
]
`;
